alacritty/src/event.rs

```diff
@@ -1106,6 +1108,26 @@ impl input::Processor<EventProxy, ActionContext<'_, Notifier, EventProxy>> {
                     WindowEvent::KeyboardInput { input, is_synthetic: false, .. } => {
                         self.key_input(input);
                     },
+                    WindowEvent::IME(ime) => match ime {
+                        // TODO decide what to do with it, but in general we should stop update of
+                        // IME input methods.
+                        IME::Enabled => {},
```

Comment:
Will it make sense to check whether ime is present based on this enabled info? We can't check based on preedit string since you don't have it right after inserting text with commit.

Reply:
What do you mean by ime is present? Like when user toggled ime? IIRC there isn't an event for that until user enter some text.

Reply:
On Wayland it comes when you enable input method. So you can clearly skip all input method handling until you've get this. For example you don't have to do anything about IME popup position, which we're currently doing in display, since there's no IME at all.

Reply:
>What do you mean by ime is present? Like when user toggled ime? IIRC there isn't an event for that until user enter some text, maybe I am wrong.

No, On Wayland/macOS you known when you're using IME. So if you don't have something like fcitx running you won't ever get Enabled.

Reply:
Enable comes from the client, and is used to signal that you have entered a text input area. If there is an input method present it will be activated and produce text input, else it will just produce plain old keyboard focus. 
So for @kchibisov's question, if I understood correctly, you can assume that input will come from either a keyboard or from an input method and never both. The compositor should handle which one you receive. You just signal that there is a possibility for text input here

Reply:
@kchibisov I do not see an enable event here? ref: [https://smithay.github.io/wayland-rs/wayland_protocols/unstable/text_input/v3/client/zwp_text_input_v3/enum.Event.html](https://smithay.github.io/wayland-rs/wayland_protocols/unstable/text_input/v3/client/zwp_text_input_v3/enum.Event.html)
Did I miss something?

Reply:
@rano-oss you won't get Enter until you have IME running, at least that's how it works for me in sway.

Reply:
Ah that is true, just got confused by the naming. Then I agree with your statement, it makes sense to check that IME is present based on this.
Also thanks, that clarified something for my compositor module.

Reply:
I think this might not apply to X11, so not sure.

Reply:
Yeah, but those platforms in winit will just send this event on window creation and emit disabled on window death, so it doesn't matter.

Reply:
> For example you don't have to do anything about IME popup position, which we're currently doing in display, since there's no IME at all.

Probably easier for us to just update IME position than to check whether we have to or not.

Reply:
>Probably easier for us to just update IME position than to check whether we have to or not.

I also feel like it, so I don't want to do anything with enabled event.